pkg/cmd/pipeline/start.go

```diff
@@ -765,6 +765,15 @@ func parsePipeline(taskLocation string, p cli.Params) (*v1beta1.Pipeline, error)
 	if err := yaml.UnmarshalStrict(b, &pipeline); err != nil {
 		return nil, err
 	}
+
+	for _, param := range pipeline.Spec.Params {
+		if param.Type == "" {
+			err := fmt.Errorf("The param '%s' does not have a type", param.Name)
```

Comment:
```suggestion
			err := fmt.Errorf("param %s does not have a type", param.Name)
```

Reply:
https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/param_types.go#L76

There are the Param types, you can check if the value is one of them or not. If not then show error and I think it is better to show error for all in once as daniel suggested.